File: EdlerCareApi/Models/Post/Post.cs
using EdlerCareApi.Models.UserProfiles;
using System.ComponentModel.DataAnnotations.Schema;

namespace EdlerCareApi.Models.Post
{
    [Table("Posts", Schema = "public")]
    public class Post : IAuditable
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public PostGenderType PostGenderType { get; set; }
        public string Body { get; set; }
        public bool IsDeleted { get; set; } = false;
        public bool Status { get; set; } = true;
        public bool IsVerified { get; set; } = false;
        public DateTimeOffset CreatedDate { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }
        public Guid CreatedBy { get; set; }
        public Guid UpdatedBy { get; set; }
        public virtual UserProfile? CreatedByName { get; set; }
        public virtual UserProfile? UpdatedByName { get; set; }
    }
}

File: ElderCareClient/src/app/core/models/Post.ts
export class Post {
  id: string;
  title: string;
  description: string;
  postGenderType: PostGenderType;
  body: string;
  isDeleted: boolean;
  createdDate?: Date;
  updatedDate?: Date;
  createdBy?: string;
  updatedBy?: string;
  isVerified: boolean;

  constructor(
    id: string,
    title: string,
    description: string,
    postGenderType: PostGenderType,
    body: string,
    isDeleted: boolean,
    createdDate: Date,
    updatedDate: Date,
    createdBy: string,
    updatedBy: string,
    isVerified: boolean
  ) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.postGenderType = postGenderType;
    this.body = body;
    this.isDeleted = isDeleted;
    this.createdDate = createdDate;
    this.updatedDate = updatedDate;
    this.createdBy = createdBy;
    this.updatedBy = updatedBy;
    this.isVerified = isVerified;
  }
}

export enum PostGenderType {
  Male = 'Male',
  Female = 'Female',
  Other = 'Other',
}
